File: dbassistant/layers/DeTokenizationLayer.py
from dbassistant.interfaces.ILayer import ILayer
from dbassistant.tokens.TokenSeries import TokenSeries
from dbassistant.tokens.Tokenizer import Tokenizer


class DeTokenizationLayer(ILayer):
    def __init__(self, tokenizer: Tokenizer):
        self.tokenizer = tokenizer

    def process(self, cargo):
        output = cargo.take()
        assert isinstance(output, TokenSeries)
        query = self.tokenizer.decode(output)
        cargo.put(query)
        return cargo

File: dbassistant/layers/DeTokenizationLayer.ts
import ILayer from '../interfaces/ILayer';
import TokenSeries from '../tokens/TokenSeries';
import Tokenizer from '../tokens/Tokenizer';
import LayerCargo from './LayerCargo';

export default class DeTokenizationLayer implements ILayer {
    private tokenizer: Tokenizer;

    constructor(tokenizer: Tokenizer) {
        this.tokenizer = tokenizer;
    }

    process(cargo: LayerCargo): LayerCargo {
        const output: TokenSeries = cargo.take();
        if (!(output instanceof TokenSeries)) {
            throw new Error('Output is not of type TokenSeries');
        }
        const query: string = this.tokenizer.decode(output);
        cargo.put(query);
        return cargo;
    }
}
